packages/x-date-pickers-pro/src/DateRangeCalendar/DateRangeCalendar.tsx

```diff
@@ -0,0 +1,553 @@
+  React.useEffect(() => {
+    const date = currentDatePosition === 'start' ? value[0] : value[1];
+    if (!date || !utils.isValid(date)) {
+      return;
+    }
+
+    const prevDate =
+      currentDatePosition === 'start' ? prevValue.current?.[0] : prevValue.current?.[1];
+    prevValue.current = value;
+
+    // The current date did not change, this call comes either from a `currentlySelectingRangeEnd` change or a change in the other date.
+    // In both cases, we don't want to change the visible month(s).
+    if (disableAutoMonthSwitching && prevDate && utils.isEqual(prevDate, date)) {
+      return;
+    }
+
+    const displayingMonthRange = calendars - 1;
+    const currentMonthNumber = utils.getMonth(calendarState.currentMonth);
+    const requestedMonthNumber = utils.getMonth(date);
+
+    if (
+      !utils.isSameYear(calendarState.currentMonth, date) ||
+      requestedMonthNumber < currentMonthNumber ||
+      requestedMonthNumber > currentMonthNumber + displayingMonthRange
+    ) {
+      const newMonth =
+        currentDatePosition === 'start'
+          ? date
+          : // If need to focus end, scroll to the state when "end" is displaying in the last calendar
+            utils.addMonths(date, -displayingMonthRange);
+
+      changeMonth(newMonth);
+    }
+  }, [currentDatePosition, value]); // eslint-disable-line
+
+  const handleSelectedDayChange = useEventCallback((newDate: TDate | null) => {
+    const { nextSelection, newRange } = calculateRangeChange({
+      newDate,
+      utils,
+      range: value,
+      currentlySelectingRangeEnd: currentDatePosition,
+    });
+
+    setCurrentDatePosition(nextSelection);
+    onCurrentDatePositionChange?.(nextSelection);
+
+    const isFullRangeSelected = currentDatePosition === 'end' && isRangeValid(utils, newRange);
+
+    setValue(newRange);
+    onChange?.(newRange, isFullRangeSelected ? 'finish' : 'partial');
+  });
+
+  const selectNextMonth = React.useCallback(() => {
+    changeMonth(utils.getNextMonth(calendarState.currentMonth));
+  }, [changeMonth, calendarState.currentMonth, utils]);
+
+  const selectPreviousMonth = React.useCallback(() => {
+    changeMonth(utils.getPreviousMonth(calendarState.currentMonth));
+  }, [changeMonth, calendarState.currentMonth, utils]);
+
+  const isNextMonthDisabled = useNextMonthDisabled(calendarState.currentMonth, {
+    disableFuture,
+    maxDate,
+  });
+
+  const isPreviousMonthDisabled = usePreviousMonthDisabled(calendarState.currentMonth, {
+    disablePast,
+    minDate,
+  });
+
+  const baseDateValidationProps: Required<BaseDateValidationProps<TDate>> = {
+    disablePast,
+    disableFuture,
+    maxDate,
+    minDate,
+  };
+
+  const commonViewProps = {
+    disableHighlightToday,
+    readOnly,
+    disabled,
+  };
+
+  // When disabled, limit the view to the selected date
+  const minDateWithDisabled = (disabled && value[0]) || minDate;
+  const maxDateWithDisabled = (disabled && value[1]) || maxDate;
```

Comment:
Should the `minDate`/`maxDate` also take into account the `disablePast`/`disableFuture`?

Reply:
After reading the all file, I better understand its purpose. Maybe renaming it `defaultizedMinDate`.

Reply:
You named those variables in `CalendarPicker` no ? :laughing: 
I can rename them in the two components

Reply:
It's not the first time (and not the last) that I introduce stupid names for internal variables 😸